spec/controllers/idv/in_person/verify_info_controller_spec.rb

```diff
@@ -83,70 +84,128 @@
-          with(hash_including(uuid_prefix: service_provider.app_id)).and_call_original
+      context 'double address verification is not enabled' do
+        let(:capture_secondary_id_enabled) { false }
+        let(:enrollment) { InPersonEnrollment.new(capture_secondary_id_enabled:) }
+        before do
+          allow(user).to receive(:establishing_in_person_enrollment).and_return(enrollment)
+        end
+        it 'sets uuid_prefix on pii_from_user' do
+          expect(Idv::Agent).to receive(:new).
+            with(hash_including(uuid_prefix: service_provider.app_id)).and_call_original
 
-        put :update
+          put :update
 
-        expect(flow_session[:pii_from_user][:uuid_prefix]).to eq service_provider.app_id
-      end
+          expect(flow_session[:pii_from_user][:uuid_prefix]).to eq service_provider.app_id
+        end
 
-      it 'passes the X-Amzn-Trace-Id to the proofer' do
-        expect_any_instance_of(Idv::Agent).to receive(:proof_resolution).
-          with(
-            kind_of(DocumentCaptureSession),
-            should_proof_state_id: anything,
-            trace_id: subject.send(:amzn_trace_id),
-            threatmetrix_session_id: nil,
-            user_id: anything,
-            request_ip: request.remote_ip,
-          )
+        context 'a user does not have an establishing in person enrollment associated with them' do
+          before do
+            allow(user).to receive(:establishing_in_person_enrollment).and_return(nil)
+          end
+
+          it 'disables double address verification for the user' do
+            expect_any_instance_of(Idv::Agent).to receive(:proof_resolution).
+              with(
+                kind_of(DocumentCaptureSession),
+                should_proof_state_id: anything,
+                trace_id: subject.send(:amzn_trace_id),
+                threatmetrix_session_id: nil,
+                user_id: anything,
+                request_ip: request.remote_ip,
+                double_address_verification: false,
+              )
+
+            put :update
+          end
+        end
 
-        put :update
-      end
+        it 'passes the X-Amzn-Trace-Id to the proofer' do
+          expect_any_instance_of(Idv::Agent).to receive(:proof_resolution).
+            with(
+              kind_of(DocumentCaptureSession),
+              should_proof_state_id: anything,
+              trace_id: subject.send(:amzn_trace_id),
+              threatmetrix_session_id: nil,
+              user_id: anything,
+              request_ip: request.remote_ip,
+              double_address_verification: false,
+            )
 
-      it 'only enqueues a job once' do
-        put :update
-        expect_any_instance_of(Idv::Agent).to_not receive(:proof_resolution)
+          put :update
+        end
 
-        put :update
-      end
+        it 'only enqueues a job once' do
+          put :update
+          expect_any_instance_of(Idv::Agent).to_not receive(:proof_resolution)
```

Comment:
so I know it already exists here, so feel free to skip for this PR, but I would love if we could follow up:

`expect_any_instance_of` is my absolute least favorite way to stub, because it's so unspecific (and sometimes leads to test pollution)

We could refactor the controller to have a memoized getter for the IDV Agent

```ruby
def idv_agent
  @idv_agent ||= Idv::Agent.new
end
```

and then here, and in the rest of the file, we could say `expect(controller.idv_agent).to_not receive(:proof_resolution)`

Reply:
Will note that I agree on avoiding `expect_any_instance_of`, but disagree on the usage of a memoized getter for stubbing (in favor of constructor injection). Since stubbing getters or memoized getters in this codebase seems standard though I will defer to Zach on that point.

Reply:
I created a [new issue](https://cm-jira.usa.gov/projects/LG/issues/LG-9814) to address this. It was suggested to address it in [LG-9440](https://cm-jira.usa.gov/browse/LG-9440). I addressed a little bug in that issue and did not want that PR to get too large.